### PR TITLE
Perf tests

### PR DESCRIPTION
**Pull request by seavan**
Cache engine implementation for SDK.

### CODE REVIEW

File: app/store/sqlcipher-db-storage.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        return JSON.deserialize(r[0].rows.item(0).value);
+    }
+
+    setValueInternal(key, value) {
+        return this.sql.executeSql(
+            'INSERT OR REPLACE INTO key_value(key, value) VALUES(?, ?)', [key, value]
+        );
+    }
+
+    async setValue(key, value, confirmUpdate) {
```
Comment (anri-asaturov) [inline app/store/sqlcipher-db-storage.js]: this whole thing needs to be a transaction, read and write are async, so another might happen in between

File: app/store/sqlcipher-db-storage.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+    async getAllKeys() {
+        const result = [];
+        const r = await this.sql.executeSql(
+            'SELECT key FROM key_value'
+        );
+        if (!r.length || !r[0].rows.length) return result;
+        const table = r[0].rows;
+        for (let i = 0; i < table.length; ++i) {
+            result.push(JSON.parse(table.item(i).key));
```
Comment (anri-asaturov) [inline app/store/sqlcipher-db-storage.js]: json parse key? Isn't it just string?

**Comment by seavan** [comment]
@anri-asaturov added transaction support

**Comment by seavan** [comment]
@anri-asaturov please review.

**Comment by ebilg** [comment]
can be merged

**Pull request merged**